File: apps/codemap-search/experiments/event-navigation-poc/language-corpus/regressions/examples/lua/controls/boundaries.lua
local Router = {}
function Router:install(f, g)
    self.cb = f -- @S1
    self.other = g -- @S2
    self.lookup = {}
    self.lookup.cb = f -- @S3
end
function Router:fire() self["cb"]() end -- @I1
function Router:map() self.lookup["cb"]() end -- @I2
return Router

File: apps/codemap-search/experiments/event-navigation-poc/language-corpus/regressions/examples/rust/controls/boundaries.rs
struct Router { cb: Box<dyn Fn()>, other: Box<dyn Fn()> }
impl Router {
    fn install(&mut self, f: Box<dyn Fn()>, g: Box<dyn Fn()>) {
        self.cb = f; // @S1
        self.other = g; // @S2
    }
    fn fire(&self) {
        let callback = self.cb.as_ref();
        callback(); // @I1
    }
}
